test/encode_test.exs

```diff
@@ -159,6 +159,33 @@ defmodule Jason.EncoderTest do
     assert to_json(%{a: 3.14159, b: 1}, pretty: false) == ~s|{"a":3.14159,"b":1}|
   end
 
+  defp to_json(value, opts \\ []) do
+    Jason.encode!(value, opts)
+  end
+
+end
+
+defmodule KeywordTester do
+  defstruct [:baz, :foo, :quux]
+end
+
+defimpl Jason.Encoder, for: [KeywordTester] do
+  def encode(struct, opts) do
+    struct
+    |> Map.from_struct
+    |> Enum.map(&(&1))
+    |> Jason.Encode.keyword(opts)
+  end
+end
+
+defmodule Jason.KeywordTest do
+  use ExUnit.Case, async: true
+
+  test "using keyword list encoding" do
+    t = %KeywordTester{baz: :bar, foo: "bag", quux: 42}
+    assert to_json(t) == ~s({"baz":"bar","foo":"bag","quux":42})
+  end
+
```

Comment:
Why the separate test module? The struct definition and `defimpl` can be nested in the main test module just fine.

Reply:
today i learned something about elixir!